docker-mender-convert

```diff
@@ -16,38 +16,34 @@
 
 set -e
 
-. modules/git.sh
-
 IMAGE_NAME=${IMAGE_NAME:-mender-convert}
+DEPLOY_DIRECTORY=${DEPLOY_DIRECTORY:-"$(pwd)/deploy"}
+INPUT_DIRECTORY=${INPUT_DIRECTORY:-"$(pwd)/input"}
+LOGS_DIRECTORY=${LOGS_DIRECTORY:-"$(pwd)/logs"}
 
-MENDER_CONVERT_DIR="$(pwd)"
-
-#Pass in version in-case we are added as a git submodule
-MENDER_CONVERT_VERSION=$(git_mender_convert_version)
+mkdir -p "$DEPLOY_DIRECTORY"
+mkdir -p "$LOGS_DIRECTORY"
 
-# Create a unique work directory so we can run multiple containers in parallel
-WORK_DIR=$(mktemp -d -p $(pwd) -t work.XXXX)
-WORK_SUFFIX=$(echo $WORK_DIR | awk -F. '{print $NF}')
-LOG_FILE="convert.log.${WORK_SUFFIX}"
+# Create a unique log file so we can run multiple containers in parallel
+LOG_FILE="convert.log.$(date +%s)-$$"
 
 set +e
 docker run \
   --rm \
-  -v $MENDER_CONVERT_DIR:/mender-convert \
-  -v $WORK_DIR:/mender-convert/work \
+  -v "$INPUT_DIRECTORY":/mender-convert/input \
+  -v "$LOGS_DIRECTORY":/mender-convert/logs \
+  -v "$DEPLOY_DIRECTORY":/mender-convert/deploy \
+  --mount type=tmpfs,destination=/mender-convert/work \
```

Comment:
`tmpfs` is not a good idea. Images can easily be bigger than all available memory, which would not work with `tmpfs`.

Is a volume even required for the work dir?

Reply:
The reason I used a volume was because they are temporary files - from my understanding of docker you should never change the internal file system during docker run - I might be wrong. Good point though about the size of the images!

Reply:
Reading it up you're right - no need for a volume here....